server/repository/src/db_diesel/document.rs

```diff
@@ -233,27 +234,27 @@ impl<'a> DocumentRepository<'a> {
     }
 
     /// Inserts a document
-    fn _insert(&self, doc: &Document) -> Result<(), RepositoryError> {
+    pub fn insert(&self, doc: &Document) -> Result<i64, RepositoryError> {
         diesel::insert_into(document::dsl::document)
             .values(doc.to_row()?)
             .execute(self.connection.lock().connection())?;
-        Ok(())
+        self.insert_changelog(doc, RowActionType::Upsert)
     }
 
-    fn toggle_is_sync_update(&self, id: &str, is_sync_update: bool) -> Result<(), RepositoryError> {
-        diesel::update(document_is_sync_update::table.find(id))
-            .set(document_is_sync_update::dsl::is_sync_update.eq(is_sync_update))
-            .execute(self.connection.lock().connection())?;
-
-        Ok(())
-    }
+    fn insert_changelog(
+        &self,
+        row: &Document,
+        action: RowActionType,
+    ) -> Result<i64, RepositoryError> {
+        let row = ChangeLogInsertRow {
+            table_name: ChangelogTableName::Document,
```

Comment:
This table_name is duplicated, so probably mean something's wrong, maybe just need one of these?